src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBufferReader.cs

```diff
@@ -373,10 +373,10 @@ public long ReadInt64()
             => BitConverter.IsLittleEndian ? ReadRawBytes<long>() : BinaryPrimitives.ReverseEndianness(ReadRawBytes<long>());
 
         public float ReadSingle()
-            => ReadRawBytes<float>();
+            => BinaryPrimitives.ReadSingleLittleEndian(GetBuffer(sizeof(float), out int offset).AsSpan(offset, sizeof(float)));
 
         public double ReadDouble()
-            => ReadRawBytes<double>();
+            => BinaryPrimitives.ReadSingleLittleEndian(GetBuffer(sizeof(double), out int offset).AsSpan(offset, sizeof(double)));
```

Comment:
Shouldn't this be ReadDoubleLittleEndian rather than ReadSingleLittleEndian?

Reply:
Yeah. I believe this is the source of the test failures.